swe_roe/ex2b.c

```diff
@@ -1759,14 +1866,31 @@ PetscErrorCode RHSFunctionForBoundaryEdges(RDyApp app, Vec F, PetscReal *amax_va
     if (cells->is_local[l] && b_ptr[l] == 0) {
       // Perform computation for a boundary edge
 
-      if (cells->is_local[l] && b_ptr[l] == 0) {
-        hr_vec_bnd[ii] = hl_vec_bnd[ii];
+      PetscReal uperp, q, velocity;
+      switch (edges->boundary_edge_types[ii]) {
+        case REFLECTING_WALL:
+          hr_vec_bnd[ii] = hl_vec_bnd[ii];
 
-        PetscReal dum1 = Square(sn_vec_bnd[ii]) - Square(cn_vec_bnd[ii]);
-        PetscReal dum2 = 2.0 * sn_vec_bnd[ii] * cn_vec_bnd[ii];
+          PetscReal dum1 = Square(sn_vec_bnd[ii]) - Square(cn_vec_bnd[ii]);
+          PetscReal dum2 = 2.0 * sn_vec_bnd[ii] * cn_vec_bnd[ii];
 
-        ur_vec_bnd[ii] = ul_vec_bnd[ii] * dum1 - vl_vec_bnd[ii] * dum2;
-        vr_vec_bnd[ii] = -ul_vec_bnd[ii] * dum2 - vl_vec_bnd[ii] * dum1;
+          ur_vec_bnd[ii] = ul_vec_bnd[ii] * dum1 - vl_vec_bnd[ii] * dum2;
+          vr_vec_bnd[ii] = -ul_vec_bnd[ii] * dum2 - vl_vec_bnd[ii] * dum1;
+          break;
+        case CRITICAL_OUTFLOW:
+          uperp = ul_vec_bnd[ii] * cn_vec_bnd[ii] + vl_vec_bnd[ii] * sn_vec_bnd[ii];
+          q     = hl_vec_bnd[ii] * fabs(uperp);
```

Comment:
Can you leave a note here to indicate perpendicular velocity is used here, while OFM uses absolute velocity? In case we want to identify the differences between two models.

Reply:
I will add a comment.